java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.se;
+
+import com.google.common.collect.Maps;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.InstanceOfTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.ParenthesizedTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TypeCastTree;
+import org.sonar.plugins.java.api.tree.UnaryExpressionTree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import javax.annotation.CheckForNull;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConstraintManager {
+
+  private final Map<Tree, SymbolicValue> map = new HashMap<>();
+  /**
+   * Map to handle lookup of fields.
+   * */
+  private final Map<Symbol, SymbolicValue> symbolMap = new HashMap<>();
+
+
+
+  public SymbolicValue createSymbolicValue(Tree syntaxNode) {
+    SymbolicValue result = map.get(syntaxNode);
+    if (result == null && syntaxNode.is(Tree.Kind.IDENTIFIER)) {
+      result = symbolMap.get(((IdentifierTree) syntaxNode).symbol());
+    } else if (result == null && syntaxNode.is(Tree.Kind.VARIABLE)) {
+      result = symbolMap.get(((VariableTree) syntaxNode).symbol());
+    }
+    if (result == null) {
+      result = new SymbolicValue.ObjectSymbolicValue(map.size()+ProgramState.EMPTY_STATE.constraints.size());
+      map.put(syntaxNode, result);
+      if(syntaxNode.is(Tree.Kind.IDENTIFIER)) {
+        symbolMap.put(((IdentifierTree) syntaxNode).symbol(), result);
+      } else if(syntaxNode.is(Tree.Kind.VARIABLE)) {
+        symbolMap.put(((VariableTree) syntaxNode).symbol(), result);
+      }
+    }
+    return result;
+  }
+
+
+  public SymbolicValue eval(ProgramState programState, Tree syntaxNode) {
+    syntaxNode = skipTrivial(syntaxNode);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "syntaxNode". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)